src/bindings/c/include/openvino/c/ov_infer_request.h

```diff
@@ -69,16 +70,116 @@ typedef struct {
 OPENVINO_C_API(ov_status_e)
 ov_infer_request_set_tensor(ov_infer_request_t* infer_request, const char* tensor_name, const ov_tensor_t* tensor);
 
+/**
+ * @brief Sets an input/output tensor to infer on.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param port  Port of the input or output tensor.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensor_by_node(ov_infer_request_t* infer_request, const ov_output_node_t* port, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets an input/output tensor to infer on.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param port  Port of the input or output tensor.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensor_by_const_node(ov_infer_request_t* infer_request, const ov_output_const_node_t* port, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets a batch of tensors for input data to infer by tensor name.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor_name  Name of the input tensor.
+ * @param tensors Input tensors for batched infer request. The type of each tensor must match the model
+ * input element type and shape (except batch dimension). Total size of tensors must match the input size.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensors(ov_infer_request_t* infer_request, const char* tensor_name, const ov_tensor_list_t* tensors);
```

Comment:
which will be better? *_set_tensors or *_set_tensor_list or *_set_batch_tensors ? To align C++ method, it should be *_set_tensors, but user is easy to make mistake for *_set_tensor() and *_set_tensors().

Reply:
ok, we keep it as *_set_tensors.

Reply:
done